File: SEP3_Client/Model/Unit/Group/ChatGroup.cs
using System.Collections.Generic;

namespace SEP3_Client.Model.Unit.Group
{
    public class ChatGroup
    {
        public string GroupId { get; set; }
        public string GroupName { get; set; }
        public string CreatorId { get; set; }
        public List<string> AccountIdList { get; set; }

        public ChatGroup(string groupName,string creatorId)
        {
            GroupId = null;
            GroupName = groupName;
            CreatorId = creatorId;
            AccountIdList = new List<string> {creatorId};
        }
        
        public ChatGroup(string groupId,ChatGroup chatGroup)
        {
            GroupId = groupId;
            GroupName = chatGroup.GroupName;
            CreatorId = chatGroup.CreatorId;
            AccountIdList = new List<string> {chatGroup.CreatorId};
        }
        
        public ChatGroup() {}

        public string AddAccountId(string id)
        {
            if (!HasId(id))
            {
                AccountIdList.Add(id);
                return null;
            }
            else
            {
                return "The [" + id + "] is in the group.";
            }
        }

        public bool HasId(string searchId)
        {
            foreach (var id in AccountIdList)
            {
                if (searchId==id)
                {
                    return true;
                }
            }
            return false;
        }

        public void RemoveAccountById(string id)
        {
            AccountIdList.Remove(id);
        }

        public ChatGroup Copy()
        {
            return new ChatGroup
            {
                GroupId = GroupId,
                CreatorId = CreatorId,
                AccountIdList = new List<string>(AccountIdList)
            };
        }
    }
}

File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/unit/group/ChatGroup.java
package com.example.SEP3_Database.model.domain.unit.group;

import java.util.ArrayList;

public class ChatGroup {
    private String GroupId;
    private String GroupName;
    private String CreatorId;
    private ArrayList<String> AccountIdList;

    public ChatGroup(String groupName,String creatorId)
    {
        GroupId = null;
        GroupName = groupName;
        CreatorId = creatorId;
        AccountIdList = new ArrayList<>();
        AccountIdList.add(creatorId);
    }

    public ChatGroup(String groupId,ChatGroup chatGroup)
    {
        GroupId = groupId;
        GroupName = chatGroup.GroupName;
        CreatorId = chatGroup.CreatorId;
        AccountIdList = new ArrayList<>();
        AccountIdList.add(CreatorId);
    }

    private ChatGroup(ChatGroup chatGroup)
    {
        GroupId = chatGroup.GroupId;
        GroupName = chatGroup.GroupName;
        CreatorId = chatGroup.CreatorId;
        AccountIdList = chatGroup.AccountIdList;
    }

    public String getGroupId() {
        return GroupId;
    }

    public String getGroupName() {
        return GroupName;
    }

    public String getCreatorId() {
        return CreatorId;
    }

    public ArrayList<String> getAccountIdList() {
        return AccountIdList;
    }

    public void setGroupId(String groupId) {
        GroupId = groupId;
    }

    public void setGroupName(String groupName) {
        GroupName = groupName;
    }

    public void setCreatorId(String creatorId) {
        CreatorId = creatorId;
    }

    public void setAccountIdList(ArrayList<String> accountIdList) {
        AccountIdList = accountIdList;
    }

    public String addAccountId(String id)
    {
        if (!hasId(id))
        {
            AccountIdList.add(id);
            return null;
        }
        else
        {
            return "The [" + id + "] is in the group.";
        }
    }

    public boolean hasId(String searchId)
    {
        for (int x=0;x<AccountIdList.size();x++)
        {
            if (searchId.equals(AccountIdList.get(x)))
            {
                return true;
            }
        }
        return false;
    }

    public void update(ChatGroup newChatGroup)
    {
        GroupId = newChatGroup.GroupId;
        GroupName = newChatGroup.GroupName;
        CreatorId = newChatGroup.CreatorId;
        AccountIdList = newChatGroup.AccountIdList;
    }

    public void removeAccountById(String id)
    {
        AccountIdList.remove(id);
    }

    public ChatGroup copy()
    {
        return new ChatGroup(this);
    }
}
